rct229/rules/section6/section6rule7.py

```diff
@@ -0,0 +1,131 @@
+from rct229.data.schema_enums import schema_enums
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all
+
+MSG_WARN_DAYLIGHT_NO_SCHEDULE = "Some of the spaces in zone are modeled with window or skylight and some of the spaces in zone are modeled with daylighting control directly through simulation. Verify if the mandatory lighting control requirements are modeled correctly in zone."
+MSG_WARN_DAYLIGHT = "Some of the spaces in zone are modeled with window or skylight and some of the spaces in zone are modeled with daylighting control with schedule. Verify if schedule adjustment is modeled correctly."
+MSG_WARN_NO_DAYLIGHT = "Some of the spaces in zone are modeled with fenestration but no daylighting controls. The design must include mandatory daylighting controls unless any of the exceptions to 90.1 section 9.4.1.1€ apply."
+
+DAYLIGHT_CONTROL_TYPE = schema_enums["LightingDaylightingControlType"].NONE
+
+
+class Section6Rule7(RuleDefinitionListIndexedBase):
+    """Rule 7 of ASHRAE 90.1-2019 Appendix G Section 6 (Lighting)"""
+
+    def __init__(self):
+        super(Section6Rule7, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, False, True),
+            each_rule=Section6Rule7.BuildingRule(),
+            index_rmr="proposed",
+            id="6-7",
+            description="Proposed building is modeled with daylighting controls directly or through schedule adjustments.",
+            rmr_context="ruleset_model_instances/0/buildings",
+        )
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section6Rule7.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, False, True),
+                each_rule=Section6Rule7.BuildingRule.ZoneRule(),
+                index_rmr="proposed",
+                list_path="$..zones[*]",
+            )
+
+        class ZoneRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(Section6Rule7.BuildingRule.ZoneRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, False, True),
+                    each_rule=Section6Rule7.BuildingRule.ZoneRule.SpaceRule(),
+                    required_fields={
+                        "$": ["surfaces"],
+                        "$.surfaces[*]": ["adjacent_to"],
+                    },
+                    index_rmr="proposed",
+                    list_path="$..spaces[*]",
+                )
+
+            def create_data(self, context, data=None):
+                zone_p = context.proposed
+
+                return {
+                    "has_daylight_control_flag": len(
```

Comment:
According to the RDS, shouldn't this be `zone_pdaylight_flag_p`?
Otherwise, I love this part of the code.

Reply:
Correct! I just realized that ```zone_pdaylight_flag_p``` should be ```daylight_flag_p``` in the RDS. Hence, I changed ```has_daylight_control_flag``` to ```daylight_flag_p```. I appreciate for finding this.